sentry/src/main/java/io/sentry/SentryClient.java

```diff
@@ -317,11 +315,39 @@ public void captureSession(final @NotNull Session session, final @Nullable Objec
     return envelope.getHeader().getEventId();
   }
 
+  @Override
+  public SentryId captureTransaction(SentryTransaction transaction, Scope scope, Object hint) {
+    Objects.requireNonNull(transaction, "Transaction is required.");
+
+    options
+        .getLogger()
+        .log(SentryLevel.DEBUG, "Capturing transaction: %s", transaction.getEventId());
+
+    Session session = null;
+
+    SentryId sentryId = transaction.getEventId();
+
+    try {
+      final SentryEnvelope envelope = buildEnvelope(transaction, session);
+
+      if (envelope != null) {
```

Comment:
if this is false, it'd still return the original `sentryId`, but it should return `SentryId.EMPTY_ID`

Reply:
👍

Reply:
Fixed